pykrige/tests/test_core.py

```diff
@@ -2013,3 +2013,87 @@ def test_ok_geometric():
 
     # Execute on grid:
     z, ss = OK.execute('grid', grid_lon, grid_lat)
+
+
+def test_ok_geographic_vs_euclid():
+    # Generate some random data close to the north pole.
+    # Then we use a polar projected 2d euclidean coordinate
+    # system and compare the kriging results in that coordinate
+    # system with the geographic-option results.
+    # If data point distance to the north pole is small enough
+    # (choose maximum 0.01 degrees), the differences due to curvature
+    # should be negligible.
+    np.random.seed(89239413)
+    from_north = 1e-2*np.random.random(5)
+    lat = 90.0-from_north
+    lon = 360.0*np.random.random(5)
+    z = np.random.random(5)
+    z -= z.mean()
+    x = from_north * np.cos(np.deg2rad(lon))
+    y = from_north * np.sin(np.deg2rad(lon))
+
+    # Generate grids:
+    grid_lon = 360.0 * np.linspace(0, 1, 50)
+    grid_from_north = np.linspace(0,0.01,10)
+    grid_lat = 90.0 - grid_from_north
+    grid_x = grid_from_north[:,np.newaxis] * np.cos(np.deg2rad(grid_lon[np.newaxis,:]))
+    grid_y = grid_from_north[:,np.newaxis] * np.sin(np.deg2rad(grid_lon[np.newaxis,:]))
+    grid_lon, grid_lat = np.meshgrid(grid_lon, grid_lat, indexing='xy')
+
+    # Flatten the grids:
+    grid_x = grid_x.flatten()
+    grid_y = grid_y.flatten()
+    grid_lon = grid_lon.flatten()
+    grid_lat = grid_lat.flatten()
+
+    # Calculate and compare distance matrices ensuring that that part
+    # of the workflow works as intended (tested: 2e-9 is currently the
+    # match for this setup):
+    d_eucl = np.sqrt(  (x[:,np.newaxis] - grid_x[np.newaxis,:])**2
+                     + (y[:,np.newaxis] - grid_y[np.newaxis,:])**2)
```

Comment:
Use  `scipy.spatial.cdist` -- it has a higher accuracy. Here there may be some accuracy loss I think.

Reply:
Should I still do it? That would mean adding a scipy include to test script.

Reply:
As you wish. We use scipy in the main package anyway. Unless it's less convenient to put the input array in the right format..